feat(global): Isolate extensions and make it compatible with cumcumber 3

bnadim opened the pull request:
<!--
Thanks for submitting a pull request!
Please provide enough information so that others can review your pull request.
The two fields below are mandatory.
-->

**Summary**
Isolated extensions context per world rather than shared module

<!--
Explain the **motivation** for making this change.
What existing problem does the pull request solve?
-->

**Test plan**

plouc commented on src/extensions/fixtures/fixtures_loader.js:
```diff
@@ -11,183 +11,214 @@ const chalk = require('chalk')
-}
+class FixturesLoader {
+    /**
+     * @param {string} _fixturesDir - The name of the fixtures directory relative to feature
```
`@param {string} [_fixturesDir='fixtures'] - The name of the fixtures directory relative to feature`

bnadim replied:
Ok

plouc commented on src/extensions/fixtures/fixtures_loader.js:
```diff
@@ -11,183 +11,214 @@ const chalk = require('chalk')
+    /**
+     * Configures the loader
+     *
+     * @param {string} _fixturesDir - The name of the fixtures directory relative to feature
```
same here

plouc replied:
You can remove the `_`, it was only there to not conflict with var name which is now bound to `this`

bnadim replied:
ok

plouc commented on src/extensions/cli/cli.js:
```diff
@@ -10,182 +10,211 @@ const { spawn } = require('child_process')
+     *
+     * @param {string} _cwd - The new CWD
+     */
+    setCwd(_cwd) {
```
You can remove the `_`, it was only there to not conflict with var name which is now bound to `this`

bnadim replied:
ok

plouc commented on src/extensions/cli/cli.js:
```diff
@@ -10,182 +10,211 @@ const { spawn } = require('child_process')
+     *
+     * @param {Object} _env - The environment variables object
+     */
+    setEnvironmentVariables(_env) {
```
You can remove the `_`, it was only there to not conflict with var name which is now bound to `this`

plouc commented on src/extensions/fixtures/fixtures_loader.js:
```diff
@@ -11,183 +11,214 @@ const chalk = require('chalk')
+     *
+     * @param {string} _featureUri - Feature uri
+     */
+    setFeatureUri(_featureUri) {
```
You can remove the `_`, it was only there to not conflict with var name which is now bound to `this`

bnadim replied:
ok

plouc commented on src/extensions/http_api/client.js:
```diff
@@ -13,230 +13,256 @@ const { Cookie } = require('tough-cookie')
+     *
+     * @param {Object} _query
+     */
+    setQuery(_query) {
```
You can remove the `_`, it was only there to not conflict with var name which is now bound to `this`

bnadim replied:
ok

plouc commented on src/extensions/http_api/client.js:
```diff
@@ -13,230 +13,256 @@ const { Cookie } = require('tough-cookie')
+     *
+     * @param {Object} _headers
+     */
+    setHeaders(_headers) {
```
You can remove the `_`, it was only there to not conflict with var name which is now bound to `this`

bnadim replied:
ok

plouc commented:
@bnadim, thank you! just a few micro adjustments.

bnadim commented:
If you can review please, i followed the reviews

bnadim commented:
@plouc i fixed what you asked, fixed compatibility issue with cumcumber 3 and update travis to support cucumber 2 and 3.
If it's ok for you please merge it when you can :)
